students/perrymant/21-docker.md

```diff
@@ -0,0 +1,134 @@
+# Docker
+
+> Docker runs processes in isolated containers. A container is a process which runs on a host. The host may be local or remote. When an operator executes docker run, the container process that runs is isolated in that it has its own file system, its own networking, and its own isolated process tree separate from the host. - [Docker](https://docs.docker.com/)
+
+- What is a container? At its simplest it's a sandbox for a process - allowing the process to have its own namespace, restricting certain capabilities and imposing resource limits as required.
+- Container technology allows multiple isolated user space instances to be run on a single host.
+
+```
++---++---++---++---++---+   \
```

Comment:
RIght now I'm a little loosed thinking what means these letters and numbers... They are there because some special naming or are just container names? If so, maybe could be better put something like this?
```
----------------------
| <c1> | <c2> | <c3> |
----------------------
| <c4> | <c5> | <c6> |
----------------------
| <c7> | <c8> | <c9> |
----------------------
```
If not, please forget this comment.

```

Reply:
They read "App" with a number, vertically. I would like to keep it like this if that's OK.

Reply:
Ahh, I see, yes now makes sense to me! I was seeing it as some kind of output of the container list command, when in really was very simple. Thanks for clarification!